### ISSUE TITLE
Artifacts in reconstructed images

### ISSUE THREAD

**Issue by hashishoya** [cameronfabbri/Underwater-Color-Correction#4]
After downloading your checkpoints (pretrained network) and generating images on your test set, the generated images include artifacts. Please check the attached images for more details. Could you please check your code or your checkpoints again? Thanks :) 
![00196_gen](https://user-images.githubusercontent.com/29784227/49832023-cca32b00-fd95-11e8-8d95-3ba013855e95.png)
![00118_gen](https://user-images.githubusercontent.com/29784227/49832032-d036b200-fd95-11e8-8024-75f00f477e57.png)
![00023_gen](https://user-images.githubusercontent.com/29784227/49832036-d331a280-fd95-11e8-9de0-0fcf5ae58db0.png)

**Comment by hashishoya** [comment]
How long did it take you to train the network? If you trained it on their same dataset, could you please share your checkpoints? Thanks

**Comment by falrom** [comment]
I haven't finished the training yet (about two days). If the test works, I will give you the ckpt. But I have some problems with the [BN layers](https://github.com/cameronfabbri/Underwater-Color-Correction/issues/5).
So, emmm....I am not sure that I will complete the final training.

**Comment by cameronfabbri** [comment]
These are interesting, thank you for pointing out some issues. I've only seem similar artifacts when evaluating on images that were far different than the training set, but the shark @falrom posted I've tested on and haven't seen artifacts. How long did you train for? My results are after 100 epochs of training.

**Comment by falrom** [comment]
@cameronfabbri The fish picture above was processed by the pretrained checkpoint downloaded from the [project page](http://irvlab.cs.umn.edu/enhancing-underwater-imagery-using-generative-adversarial-networks). I retrained the network and got good results like this:
![89500_paper2_gen](https://user-images.githubusercontent.com/18463741/50336123-4bdbf000-0548-11e9-98fd-4b01c0493109.png)
I don't know why... (Mine: python2.7 + TF1.10.0)

@hashishoya I have finished the training and [this is the checkpoint](https://drive.google.com/file/d/10k7cHiPlS8vQHvkkHoFTSh4outvE3-Xt/view?usp=sharing). It took about 3 days. (step 89500/epoch 468)
![cut](https://user-images.githubusercontent.com/18463741/50337322-b0e51500-054b-11e9-973b-41c37b4447cc.PNG)
It seems that 100 epochs are enough.

**Comment by hashishoya** [comment]
Can you please tell what GPU device model you used? Thanks for sharing your checkpoints :)

**Comment by Saulzar** [comment]
I see these same artifacts - but then if I use the checkpoint you provided it has even worse artifacts. I evaluated it using the 'eval_one.py' script. Any idea what might be wrong?

![l00006130_real](https://user-images.githubusercontent.com/1128677/50570572-6427f680-0df5-11e9-9440-4b0c874c1071.png)
![l00006130_gen](https://user-images.githubusercontent.com/1128677/50570573-67bb7d80-0df5-11e9-8627-1dfce0992848.png)

**Comment by Saulzar** [comment]
Here's the output using the model from the author showing the same artifacts.

![l00006130_real_gen](https://user-images.githubusercontent.com/1128677/50570578-920d3b00-0df5-11e9-9eda-27cabf69e73b.png)

**Comment by Saulzar** [comment]
I can't seem to get tensorflow to load the checkpoint provided from @falrom - it does not contain the file called 'checkpoint' in the directory, do you know what the story is there? 

'tf.train.get_checkpoint_state(EXPERIMENT_DIR)' returns None

The checkpoint provided by the author @cameronfabbri loads correctly but I see the same artifacts.

![l00007680_gen](https://user-images.githubusercontent.com/1128677/50570633-2926c280-0df7-11e9-95b4-9338d1674610.png)

![l00006130_gen](https://user-images.githubusercontent.com/1128677/50570636-3ba0fc00-0df7-11e9-9b34-f514bcc40c95.png)

**Comment by falrom** [comment]
You can just assign the value to EXPERIMENT_DIR directly in the python code.
```python
    # EXPERIMENT_DIR  = 'checkpoints/LOSS_METHOD_'+LOSS_METHOD\
    #                   +'/NETWORK_'+NETWORK\
    #                   +'/L1_WEIGHT_'+str(L1_WEIGHT)\
    #                   +'/IG_WEIGHT_'+str(IG_WEIGHT)\
    #                   +'/AUGMENT_'+str(AUGMENT)\
    #                   +'/DATA_'+DATA+'/'\

    EXPERIMENT_DIR = 'checkpoints/LOSS_METHOD_wgan/NETWORK_pix2pix/L1_WEIGHT_100.0/IG_WEIGHT_0.0/AUGMENT_0/DATA_underwater_imagenet/'
```
The EXPERIMENT_DIR  must contain these 3 files:
![image](https://user-images.githubusercontent.com/18463741/50579682-45237600-0e82-11e9-85c7-2661fce19bd5.png)

**Comment by Saulzar** [comment]
I solved it by adding a file called 'checkpoint' which cointains this:

> model_checkpoint_path: "checkpoint-89500"
> all_model_checkpoint_paths: "checkpoint-89000"
> all_model_checkpoint_paths: "checkpoint-89500"


After that it works - I had a play around trying it on some real underwater images, unfortunately it doesn't seem to produce nice results with significant turbidity, with a lot of 'gridding' going on. I guess removing turbidity is a whole different problem!

![r00010550_real](https://user-images.githubusercontent.com/1128677/50589182-851e4380-0eea-11e9-8ecc-4bd920afbfda.png)

![r00010550_gen](https://user-images.githubusercontent.com/1128677/50589189-8a7b8e00-0eea-11e9-8621-6ae3622081ea.png)

![l00011810_real](https://user-images.githubusercontent.com/1128677/50589196-92d3c900-0eea-11e9-8a2d-852fc389a4d5.png)

![l00011810_gen](https://user-images.githubusercontent.com/1128677/50589201-9a936d80-0eea-11e9-8f38-cbbd19c942b8.png)

**Comment by RaghuRajRai** [comment]
> @cameronfabbri The fish picture above was processed by the pretrained checkpoint downloaded from the [project page](http://irvlab.cs.umn.edu/enhancing-underwater-imagery-using-generative-adversarial-networks). I retrained the network and got good results like this:
> ![89500_paper2_gen](https://user-images.githubusercontent.com/18463741/50336123-4bdbf000-0548-11e9-98fd-4b01c0493109.png)
> I don't know why... (Mine: python2.7 + TF1.10.0)
> 
> @hashishoya I have finished the training and [this is the checkpoint](https://drive.google.com/file/d/10k7cHiPlS8vQHvkkHoFTSh4outvE3-Xt/view?usp=sharing). You may try it. It took about 3 days. (step 89500/epoch 468)
> ![cut](https://user-images.githubusercontent.com/18463741/50337322-b0e51500-054b-11e9-973b-41c37b4447cc.PNG)
> It seems that 100 epochs are enough.

@cameronfabbri  - Can you please confirm if the uploaded checkpoint performs as per the model's expectations?

**Comment by dingyan1478** [comment]
@

> You can just assign the value to EXPERIMENT_DIR directly in the python code.
> 
> ```python
>     # EXPERIMENT_DIR  = 'checkpoints/LOSS_METHOD_'+LOSS_METHOD\
>     #                   +'/NETWORK_'+NETWORK\
>     #                   +'/L1_WEIGHT_'+str(L1_WEIGHT)\
>     #                   +'/IG_WEIGHT_'+str(IG_WEIGHT)\
>     #                   +'/AUGMENT_'+str(AUGMENT)\
>     #                   +'/DATA_'+DATA+'/'\
> 
>     EXPERIMENT_DIR = 'checkpoints/LOSS_METHOD_wgan/NETWORK_pix2pix/L1_WEIGHT_100.0/IG_WEIGHT_0.0/AUGMENT_0/DATA_underwater_imagenet/'
> ```
> 
> The EXPERIMENT_DIR must contain these 3 files: ![image](https://user-images.githubusercontent.com/18463741/50579682-45237600-0e82-11e9-85c7-2661fce19bd5.png)

@falrom 
您好，不知道您方不方便告诉我怎样测试UGAN呢？我一直出现下面的错误，但是不知道该怎样改正？
![image](https://user-images.githubusercontent.com/74007964/136967481-7bb6efaf-3e07-425e-a908-44735f343a69.png)
